File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        make_heap(nums.begin(), nums.end());
        int i=-1;
        while(++i<k-1){
            pop_heap(nums.begin(), nums.end());
            nums.pop_back();
        }
        return nums.front();
    }
};

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution:

    def findKthLargest(self, nums: List[int], k: int) -> int:
        n = len(nums)

        if n <= 1:
            return nums[0]

        pivot = nums[0]

        left, right = [], []
        for i in range(1, n):
            if nums[i] < pivot:
                left.append(nums[i])
            else:
                right.append(nums[i])

        if len(right) + 1 < k:
            return self.findKthLargest(left, k-len(right)-1)
        elif len(right) + 1 > k:
            return self.findKthLargest(right, k)
        else:
            return pivot
